include/zephyr/pm/pm.h

```diff
@@ -53,16 +53,23 @@ extern "C" {
  */
 struct pm_notifier {
 	sys_snode_t _node;
+
 	/**
-	 * Application defined function for doing any target specific operations
+	 * Value indicating whether the callback should trigger on entry or
+	 * exit of a power state.
+	 */
+	enum pm_state_direction direction;
+
+	/**
+	 * Application or driver defined function for doing any target specific operations
 	 * for power state entry.
 	 */
-	void (*state_entry)(enum pm_state state);
+	void (*callback)(enum pm_state_direction direction, const void *ctx);
```

Comment:
nit: document callback arguments

also, consting context may be unnecessarily restrictive.

Reply:
Good point, will add argument documentation.

The context is const for the use case of a driver of passing a device struct.

Reply:
yep ... you keep the argument const in the internal struct and remove the const when calling the callback,

Reply:
I understand that, but doesn't the potential for undefined behavior make this problematic? Though I suppose in a driver we'd immediately cast ctx into a `const struct device *`.